plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1140,15 +1143,14 @@ public boolean downloadToFile(String urlString, Plugin plugin, @CheckForNull Fil
                     .setRetryHandler(new DefaultHttpRequestRetryHandler(maxRetries, true))
                     .build()) {
                 HttpClientContext context = HttpClientContext.create();
-                HttpHead httphead = new HttpHead(urlString);
-                try (CloseableHttpResponse ignored = httpclient.execute(httphead, context)) {
-                    HttpHost target = context.getTargetHost();
-                    List<URI> redirectLocations = context.getRedirectLocations();
-                    // Expected to be an absolute URI
-                    URI location = URIUtils.resolve(httphead.getURI(), target, redirectLocations);
-                    logVerbose(String.format("downloading %s from %s", plugin.getName(), location));
-                    FileUtils.copyURLToFile(location.toURL(), pluginFile);
-                } catch (URISyntaxException | IOException e) {
+                CredentialsProvider credentialsProvider = getCredentialsProvider();
+                if (credentialsProvider != null) {
```

Comment:
Fixed in https://github.com/jenkinsci/plugin-installation-manager-tool/pull/292/commits/1be02a1c57e484eee701f76f62543bd05815e107. Now it can really be null in case no credentials have been set.